File: upgrade/upg.go
package upg

import (
	"fmt"
	"runtime"

	"github.com/secman-team/shell"
)

upgrade := `
	l=$(curl --silent "https://api.github.com/repos/secman-team/secman/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
	c=$(secman verx)
	smLoc="/usr/local/bin"

	if [ $l == $c ]; then
		echo "secman is already up-to-date and it's the latest release $l"

	elif [ $l != $c ]; then
		sudo rm -rf $smLoc/secman*
		sudo rm -rf $smLoc/cgit*
		sudo rm -rf $smLoc/verx*

		curl -fsSL https://secman-team.github.io/install.sh | bash

		if [ -x "command -v $(secman)" ]; then
			echo "secman was upgraded successfully 🎊"
		fi
	fi
`

func Upgrade() {
	if runtime.GOOS == "windows" {
		// shell.PWSLCmd("& ~/sm/vx.ps1 --upg")
		fmt.Println("run sm-upg start")
	} else {
		shell.ShellCmd(upgrade)
	}
}

File: upgrade/upg.rb
require "rbconfig"
require "colorize"

$l = `verx secman-team/secman -l`
$c = `secman verx`

$smLoc = "/usr/local/bin"

def pre_upgrade
    system("sudo rm -rf #{$smLoc}/secman*")
    system("sudo rm -rf #{$smLoc}/cgit*")
    system("sudo rm -rf #{$smLoc}/verx*")
    system("sudo rm -rf /home/sm")
end

def core
    system("curl -fsSL https://secman-team.github.io/install/install.sh | bash")
end

sm = "secman".cyan

if $l == $c
    al = "is already up-to-date and it's the latest release"
    puts "#{sm} #{al} #{$l.yellow}"
    
elsif $l != $c
    pre_upgrade()
    core()

    puts "#{sm} was upgraded successfully"
end
